presto-main/src/main/java/io/prestosql/util/FinalizerService.java

```diff
@@ -82,15 +79,14 @@ public void addFinalizer(Object referent, Runnable cleanup)
     {
```

Comment:
I see that `addFinalizer` is used only with `PartitionedSplitCountTracker`. I wonder if it couldn't be closed explicitly